scipy/spatial/_geometric_slerp.py

```diff
@@ -0,0 +1,226 @@
+    recognizable on the unit circle:
+    >>> ax.scatter(result[...,0], result[...,1], c='k')
+    >>> circle = plt.Circle((0, 0), 1, color='grey')
+    >>> ax.add_artist(circle)
+    >>> plt.show()
+
+    Attempting to interpolate between antipodes on a circle is
+    ambiguous because there are two possible paths, and on a
+    sphere there are infinite possible paths on the geodesic surface:
+
+    >>> opposite_pole = np.array([-1, 0])
+    >>> try:
+    ...     geometric_slerp(start,
+    ...                     opposite_pole,
+    ...                     t_vals)
+    ... except ValueError:
+    ...     print("ValueError")
+    ValueError
+
+
+    Extend the original example to a sphere and plot interpolation
+    points in 3D:
+
+    >>> from mpl_toolkits.mplot3d import proj3d
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111, projection='3d')
+
+    Plot the unit sphere for reference (optional):
+
+    >>> u = np.linspace(0, 2 * np.pi, 100)
+    >>> v = np.linspace(0, np.pi, 100)
+    >>> x = np.outer(np.cos(u), np.sin(v))
+    >>> y = np.outer(np.sin(u), np.sin(v))
+    >>> z = np.outer(np.ones(np.size(u)), np.cos(v))
+    >>> ax.plot_surface(x, y, z, color='y', alpha=0.1)
+
+    Interpolating over a larger number of points
+    may provide the appearance of a smooth curve on
+    the surface of the sphere, which is also useful
+    for discretized integration calculations on a
+    sphere surface:
+
+    >>> start = np.array([1, 0, 0])
+    >>> end = np.array([0, 0, 1])
+    >>> t_vals = np.linspace(0, 1, 200)
+    >>> result = geometric_slerp(start,
+    ...                          end,
+    ...                          t_vals)
+    >>> ax.plot(result[...,0],
+    ...         result[...,1],
+    ...         result[...,2],
+    ...         c='k')
+    >>> plt.show()
+    """
+
+    start = np.asarray(start, dtype=np.float64)
+    end = np.asarray(end, dtype=np.float64)
+
+    if start.ndim != 1 or end.ndim != 1:
+        raise ValueError("Start and end coordinates "
+                         "must be one-dimensional")
+
+    if start.size != end.size:
+        raise ValueError("The dimensions of start and "
+                         "end must match (have same size)")
+
+    if start.size == 0 or end.size == 0:
+        raise ValueError("One or both input coordinates are empty")
+
+    if np.array_equal(start, end):
+        raise ValueError("Start and end coordinates cannot be the same")
+
+    if start.size == 1 or end.size == 1:
+        raise ValueError("Cannot interpolate the 0-sphere")
+
+    if not isinstance(tol, float):
+        raise ValueError("tol must be a float")
+    else:
+        tol = np.fabs(tol)
+
+    coord_dist = euclidean(start, end)
+
+    # diameter of 2 within tolerance means antipodes, which is a problem
+    # for all unit n-spheres (even the 0-sphere would have an ambiguous path)
+    if np.allclose(coord_dist, 2.0, rtol=0, atol=tol):
+        raise ValueError("start and end are antipodes"
```

Comment:
I don't know --- should it be only a warning about possible ambiguity? Can `tol` be determined automatically? (Possibly much lower than `1e-7`, on the order of `1e-15` I suppose.)

Reply:
No, it is not appropriate to emit a warning here---antipodes are **degenerate** input, the result isn't even on the unit sphere---if I disable the guard:

```python
import numpy as np
import scipy
from scipy.spatial import geometric_slerp
import matplotlib
matplotlib.use('Agg')
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import proj3d

path = geometric_slerp(start=[0, 0, 1],
                         end=[0, 0, -1],
                         t=np.linspace(0, 1, 50))

fig = plt.figure()
ax = fig.add_subplot(111, projection='3d')

# plot the unit sphere for reference ----
u = np.linspace(0, 2 * np.pi, 100)
v = np.linspace(0, np.pi, 100)
x = np.outer(np.cos(u), np.sin(v))
y = np.outer(np.sin(u), np.sin(v))
z = np.outer(np.ones(np.size(u)), np.cos(v))
ax.plot_surface(x, y, z, color='y', alpha=0.1)
# end plot unit sphere for reference ----

ax.plot_surface(x, y, z, color='y', alpha=0.1)
ax.scatter(path[..., 0],
           path[..., 1],
           path[..., 2])
ax.azim = 50
ax.elev = 80

# antipodes are degenerate input---the path is not
# even on the unit sphere:
print(np.linalg.norm(path, axis=1))

fig.savefig('antipodal_paths.png',
            dpi=300)
```

```
[1.         0.99794539 0.99179001 0.98155916 0.96729486 0.94905575
 0.92691676 0.90096887 0.8713187  0.8380881  0.80141362 0.76144596
 0.71834935 0.67230089 0.6234898  0.57211666 0.51839257 0.46253829
 0.40478334 0.34536505 0.28452759 0.22252093 0.1595999  0.09602303
 0.03205158 0.03205158 0.09602303 0.1595999  0.22252093 0.28452759
 0.34536505 0.40478334 0.46253829 0.51839257 0.57211666 0.6234898
 0.67230089 0.71834935 0.76144596 0.80141362 0.8380881  0.8713187
 0.90096887 0.92691676 0.94905575 0.96729486 0.98155916 0.99179001
 0.99794539 1.        ]
```

![antipodal_paths](https://user-images.githubusercontent.com/7903078/69012549-4d6a8f80-0934-11ea-8463-87d2600af6e9.png)

Now, if we move away from antipodes with `start=[0, np.sqrt(2)/2, np.sqrt(2)/2]`:

```
[1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1. 1.
 1. 1.]
```

![antipodal_paths](https://user-images.githubusercontent.com/7903078/69012593-b5b97100-0934-11ea-9ccc-d41792639bc7.png)

If we were to coerce Peter's algorithm to produce something less pathological for antipodes, I still think it is a bad idea---clearly defining what is degenerate input is actually helpful for computational geometry algorithms.

If an end user wants a bunch of random great circle arcs between poles they are free to opt in **explicitly** by choosing random points on a circle and then use `geometric_slerp` to define non-ambiguous paths from the poles to the equators & then fusing those data sets.

Reply:
There was a silly bug which causes this pathological output. See my comment above.

With this in mind, I think that raising an exception for degenerate input is too strict. I would go with a warning at most.

Reply:
I've changed it to a `UserWarning`, though I still disagree here. There's an infinite set of valid return values/paths now. Here is a plot of one of them, which is at least on the unit sphere now:

![antipodal_paths](https://user-images.githubusercontent.com/7903078/74704242-d65fc400-51cc-11ea-84ac-d55ecf78a03a.png)

Anyway, I'm not going to win this battle, so I've applied the patch & will mark as resolved.